keps/sig-cli/2551-return-code-normalization/README.md

```diff
@@ -0,0 +1,421 @@
+call to the old checkErr or the new function according to the value of the Environment Variable described in
+[Risks and Mitigations](#risks-and-mitigations) 
+
+### Creating new error parser functions
+
+Another design solution is to create helper functions for each steps:
+* When running validation, call cmdutil.CheckValidationErr(err) and exits with some well defined client error code, mapped to ErrorCodeClient
+* When executing, delegate the returning error to a new function (cmdutil.CheckExecutionErr) that
+can assess if the error contains some APIError (like forbidden, not found) or Client Error and return the proper
+error. Any new Return Code from execution step should be added to errors.go and the case predicted here. 
+
+The pro of this approach is that we can re-develop everything controlling the behavior.
+
+The con of this approach is that it takes much more time and code change to point every command to the 
+right error checking function.
+
+For the rollout of this approach, the new functions will call the old `CheckErr` 
+according to the value of the Environment Variable described in [Risks and Mitigations](#risks-and-mitigations), 
+or will follow the new flow.
+
+### Hybrid approach
+
+There's an Hybrid approach that can be used:
+
+* For the steps that run on the client side, create a new function that does an early exit/return with 
+an well known exit code that will be used for all client side operations (no difference between yaml
+validation, missing flag, etc)
+* For the execution step, call CheckErr, that might delegate the error validation to a new function or follow
+with the old behavior, depending on the Environment Variable described above
+
+```
+<<[UNRESOLVED external commands ]>>
+For commands that call external commands (diff, plugins, edit) this needs to be discussed.
+<<[/UNRESOLVED]>>
+```
+
+
+### Test Plan
+
+* Add Unit tests for each specific error case mapped on the error matrix
+* Add Unit tests on the commands to verify for specific cases (plugins, diff) if the right exit code is returned
+* Add e2e tests to verify if the right exit code is returned
+
+
+### Graduation Criteria
+
+#### Alpha 
+
+- Feature implemented and enabled only with Environment Variable
+
+#### Beta
+- Feature enabled by default, but enablement available via Environment Variable
+- Maximum of 2 issues regarding kubectl exit code opened while feature was in Alpha (last release)
+
+#### GA 
+- Feature enabled by default and enablement not configurable
+- No issues regarding kubectl exit code opened on the last 2 releases while feature was in Beta
+
+### Upgrade / Downgrade Strategy
+
+This behavior is feature gated through an Environment Variable and can be enabled or 
+disabled without impact to upgrade or downgrade between versions 
+
+### Version Skew Strategy
+
+Not applicable, as this is just a kubectl change
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
```

Comment:
I think you have an older version of the questions, please review the latest template and include any new questions as well. Thanks.